src/System.IO.Pipelines/System/IO/Pipelines/PreservedBuffer.cs

```diff
@@ -6,11 +6,11 @@ namespace System.IO.Pipelines
     /// <summary>
     /// Represents a buffer that can read a sequential series of bytes.
     /// </summary>
-    public struct PreservedBuffer : IDisposable
+    public readonly struct PreservedBuffer : IDisposable
     {
-        private ReadableBuffer _buffer;
+        private readonly ReadableBuffer _buffer;
 
-        internal PreservedBuffer(ref ReadableBuffer buffer)
+        internal PreservedBuffer(in ReadableBuffer buffer)
         {
             _buffer = buffer;
```

Comment:
Should we assert that the ReadableBuffer is references a BufferSegment and not an Array?

Reply:
For now, there is `Preserve` implementation for array-based readable buffers, it clones an array and while it doesn't specifically require freeing it can support the same pattern.